props-core/src/main/java/sh/props/source/Source.java

```diff
@@ -114,6 +109,28 @@ public void refresh() {
     sendLayerUpdate(Collections.unmodifiableMap(this.get()));
   }
 
+  /**
+   * Provides a unique reference for each Source implementation. By default, this method returns the
+   * implementation's name and a unique hash code computed at runtime. This method can help users
+   * reference a source in {@link sh.props.Registry#get(String, Converter, String)}.
+   *
+   * <p>This method is final and cannot be overwritten since that would create potential for
+   * confusion. When a user desired to reference sources by a deterministic identifier (reproducible
+   * in subsequent JVM runs), they should alias {@link Layer}s via {@link
+   * sh.props.RegistryBuilder#withSource(Source, String)}.
+   *
+   * @return a unique reference that's meant to enable client code to identify sources in a {@link
+   *     sh.props.Registry}, allowing users to retrieve a value directly from a source
+   */
+  public final String id() {
```

Comment:
but a default alias method is provided, for the 99% use-case, where client code does not need to reference explicit layers directly.